src/NServiceBus.NHibernate/TimeoutPersisters/OptimizedSchemaUpdate.cs

```diff
@@ -101,17 +105,27 @@ public void Execute(Action<string> scriptAction, bool doUpdate)
                 {
                     typeof(MsSql2005Dialect).FullName,
                     typeof(MsSql2008Dialect).FullName,
-                    typeof(MsSql2012Dialect).FullName,
+                    typeof(MsSql2012Dialect).FullName
                 };
 
+                var shouldOptimizeTimeoutEntityIndexes = timeoutEntityMapping != null;
+
                 for (var j = 0; j < updateSQL.Length; j++)
                 {
                     var sql = updateSQL[j];
 
-                    if (dialectScopes.Contains(dialect.GetType().FullName))
+                    if (dialectScopes.Contains(dialect.GetType().FullName) && shouldOptimizeTimeoutEntityIndexes)
                     {
-                        sql = sql.Replace("primary key (Id)", "primary key nonclustered (Id)");
-                        sql = sql.Replace("create index TimeoutEntity_EndpointIdx on TimeoutEntity (Time, Endpoint)", "create clustered index TimeoutEntity_EndpointIdx on TimeoutEntity (Endpoint, Time)");
+                        var qualifiedTimeoutEntityTableName = timeoutEntityMapping.Table.GetQualifiedName(dialect);
+
+                        if (sql.StartsWith($"create table {qualifiedTimeoutEntityTableName}"))
+                        {
+                            sql = sql.Replace("primary key (Id)", "primary key nonclustered (Id)");
+                        }
+                        else if (sql.StartsWith("create index TimeoutEntity_EndpointIdx"))
```

Comment:
@MarcinHoppe Might we want to make TimeoutEntity_EndpointIdx a constant that is also used in the mapping?